### PR TITLE
Fix code lens count to match reference resolver (issue #637)

### PR DESCRIPTION
**Pull request by bitwisecook**
## Summary

Fixes issue #637 where code lens reference counts could disagree with the "Find All References" results. The lens count is now derived from the same namespace-aware reference resolver that backs the peek/Find All References feature, ensuring they never drift.

### Key Changes

1. **Code Lens Reference Counting** (`code_lens.rs`)
   - Replaced the name-only `count_references()` function with calls to the namespace-aware `crate::references::references()` resolver
   - The lens now uses the exact same resolution logic as Find All References, positioned at the proc's name span
   - Removed the old `count_references()` function entirely
   - Added comprehensive test coverage for the fix, including edge cases like forward references, qualified calls, and same-named procs in different namespaces

2. **Semantic Tokens: Structural Keywords** (`semantic_tokens.rs`)
   - Added `KeywordArg` override to highlight structural keyword words (`if`'s `then`/`elseif`/`else`, `try`'s `on`/`trap`/`finally`) as keywords rather than strings
   - These keywords sit at argument positions (not the command-name slot), so they need explicit role marking
   - Properly handles quoted keywords by offsetting past delimiters
   - Added tests verifying keyword highlighting for `if`/`elseif`/`else` and `try`/`on`/`finally`

3. **Registry: Keyword Role Support** (`arg_role.rs`, `if_.rs`, `try_.rs`)
   - Added `ArgRole::Keyword` enum variant to mark structural keyword positions
   - Updated `if` and `try` command specs to emit `Keyword` roles for their structural keywords
   - Updated Python FFI binding to support the new role

4. **Inlay Hints: Default-Off Contract** (`lib.rs`)
   - Split `inlayHints` into two independent families: `inlayTypeHints` and `inlayParameterHints` (PR #643)
   - Inlay hints now default **off** (matching Python server behavior)
   - The retired `inlayHints` key is accepted as a backward-compatible alias for `inlayTypeHints` only
   - Added `inlay_parameter_hints_enabled()` check that returns an empty list (not `null`) when disabled
   - Added tests verifying the feature toggle behavior and backward compatibility

5. **E2E Tests**
   - Added `inlay_hint_smoke.rs` to verify the default-off contract: requests against hintable content return an empty list until explicitly enabled
   - Updated Python test fixture to use the new split keys

## Validation

- Added 8 new unit tests in `code_lens.rs` covering forward references, qualified calls, namespace scoping, and ambiguous resolution
- Added 4 new unit tests in `semantic_tokens.rs` for keyword highlighting
- Added 2 new unit tests in `arg_role.rs` for keyword role resolution
- Added 3 new unit tests in `lib.rs` for feature toggle behavior
- Added end-to-end LSP test (`inlay_hint_smoke.rs`) verifying default-off contract
- Updated Python test fixture configuration

https://claude.ai/code/session_0126QRLXCX9bZDUVDAu4AWby

### CODE REVIEW

File: rust/tcl-lsp-server/src/lib.rs [hunk 1]
```diff
@@ -673,8 +678,19 @@ impl FeatureToggles {
...
     /// Merge an editor-supplied `features` object, setting only the
     /// keys it carries (absent keys keep their last-applied value).
+    ///
+    /// The retired `inlayHints` key is a backward-compatible alias for
+    /// `inlayTypeHints` (it enables the useful variable-type hints, not
+    /// the verbose parameter-name hints; PR #643).  It is applied first
+    /// so an explicit new `inlayTypeHints` in the same object wins.
     fn apply(&mut self, features: &serde_json::Map<String, serde_json::Value>) {
+        if let Some(flag) = features.get("inlayHints").and_then(serde_json::Value::as_bool) {
+            self.set.insert("inlayTypeHints".to_owned(), flag);
```
[the remark this answers is not in the document]
Comment (bitwisecook) [reply rust/tcl-lsp-server/src/lib.rs]: Fixed in 7de24441. The legacy `inlayHints` alias now maps to `inlayParameterHints` — the family the Rust provider actually implements and gates on — so the existing editor setting keeps producing hints instead of going dead. This deliberately diverges from the Python #643 legacy→type mapping: Rust has no type-hint family, so mapping the legacy key to type would leave it a no-op here. Covered by the `legacy_inlay_hints_key_maps_to_implemented_parameter_family` unit test.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: rust/tcl-lsp-server/src/lib.rs [hunk 2]
```diff
@@ -646,7 +646,12 @@ impl FeatureToggles {
         "rename",
         "signatureHelp",
         "workspaceSymbols",
-        "inlayHints",
+        // Inlay hints split into two independently-toggled families
+        // (PR #643): inferred-type hints and parameter-name hints.  The
+        // retired `inlayHints` key is still accepted on input as a
+        // backward-compatible alias for `inlayTypeHints` (see `apply`).
+        "inlayTypeHints",
+        "inlayParameterHints",
```
[the remark this answers is not in the document]
Comment (bitwisecook) [reply rust/tcl-lsp-server/src/lib.rs:655]: Fixed in 7de24441. Added a `DEFAULT_OFF` set so both `is_enabled` and `resolved_map` fall back to `false` for `inlayTypeHints`/`inlayParameterHints`, matching the handler's default-off gate. `getEffectiveConfig` (and an exported `config.ini`) now report them disabled until an editor sets them, so a round-trip can't silently enable them. Covered by the `inlay_keys_default_off` test.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: rust/tcl-lsp-core/src/code_lens.rs [hunk 3]
```diff
@@ -77,8 +77,23 @@ pub fn code_lenses(
...
+        let name_pos = line_index.position_at_utf16(proc_def.name_span.start(), source);
+        let mut count = crate::references::references(
+            source,
+            dialect,
+            name_pos.line,
+            name_pos.character,
+            analysis,
+            false,
+        )
+        .len();
```
[the remark this answers is not in the document]
Comment (bitwisecook) [reply rust/tcl-lsp-core/src/code_lens.rs]: Fixed in 7de24441. Extracted `references::proc_reference_spans` — the namespace-aware call-site matching shared with Find All References — which takes the resolved proc directly. The code-lens loop now calls it per proc instead of `references()`, so it no longer rebuilds a `LineIndex` or rescans the proc table for each definition; it just counts the returned spans. The shared helper also keeps the lens count and the peek from drifting.

---
_Generated by [Claude Code](https://claude.ai/code)_
